src/Features/Core/Portable/StackTraceExplorer/ParsedStackFrame.cs

```diff
@@ -92,157 +75,184 @@ internal sealed class ParsedStackFrame : ParsedFrame
             }
 
-            return OriginalText[(ArgsSpan.End + 1)..FileSpan.Start];
+            return true;
         }
 
-        internal (Document? document, int line) GetDocumentAndLine(Solution solution)
+        private static bool MatchTypeArguments(ImmutableArray<ITypeSymbol> typeArguments, StackFrameTypeArgumentList? stackFrameTypeArgumentList)
         {
-            var fileMatches = GetFileMatches(solution, out var lineNumber);
-            if (fileMatches.IsEmpty)
+            if (stackFrameTypeArgumentList is null)
             {
-                return (null, 0);
+                return typeArguments.IsDefaultOrEmpty;
             }
 
-            return (fileMatches.First(), lineNumber);
+            if (typeArguments.IsDefaultOrEmpty)
+            {
+                return false;
+            }
+
+            var stackFrameTypeArguments = stackFrameTypeArgumentList.TypeArguments;
+            return typeArguments.Length == stackFrameTypeArguments.Length;
         }
 
-        /// <summary>
-        /// If the <see cref="FileSpan"/> exists it attempts to map the file path that exists
-        /// in that span to a document in the solution. It's possible the file path won't match exactly,
-        /// so it does a best effort to match based on name.
-        /// </summary>
-        private ImmutableArray<Document> GetFileMatches(Solution solution, out int lineNumber)
+        private static bool MatchType(ITypeSymbol type, StackFrameTypeNode stackFrameType)
         {
-            var fileText = OriginalText[FileSpan.Start..FileSpan.End];
-            var regex = new Regex(@"(?<fileName>.+):(line)\s*(?<lineNumber>[0-9]+)");
-            var match = regex.Match(fileText);
-            Debug.Assert(match.Success);
-
-            var fileNameGroup = match.Groups["fileName"];
-            var lineNumberGroup = match.Groups["lineNumber"];
-
-            lineNumber = int.Parse(lineNumberGroup.Value);
-
-            var fileName = fileNameGroup.Value;
-            Debug.Assert(!string.IsNullOrEmpty(fileName));
+            if (type is IArrayTypeSymbol arrayType)
+            {
+                if (stackFrameType is not StackFrameArrayTypeNode arrayTypeNode)
+                {
+                    return false;
+                }
 
-            var documentName = Path.GetFileName(fileName);
-            var potentialMatches = new HashSet<Document>();
+                ITypeSymbol currentType = arrayType;
 
-            foreach (var project in solution.Projects)
-            {
-                foreach (var document in project.Documents)
+                // Iterate through each array expression and make sure the dimensions
+                // match the element types in an array.
+                // Ex: string[,][] 
+                // [,] is a 2 dimension array with element type string[]
+                // [] is a 1 dimension array with element type string
+                foreach (var arrayExpression in arrayTypeNode.ArrayExpressions)
```

Comment:
nit: calling thse ArrayExpressions is a bit wonky.

Reply:
Thoughts on a better name? Could also be `ArrayRankSpecifiers` since it's just an array of those.

Reply:
that sounds perfect to me :)

Reply:
it also matches ArrayRankSpecifierSyntax in C#.  and ArrayTypeSyntax has RankSpecifiers :)